src/Build/Logging/FancyLogger/FancyLoggerBuffer.cs

```diff
@@ -127,31 +93,32 @@ public static void Render()
             if (Lines.Count == 0) return;
 
             // Iterate over lines and display on terminal
-            // TODO: Delimit range to improve performance 
+            string contents = string.Empty;
             int accumulatedLineCount = 0;
             foreach (FancyLoggerBufferLine line in Lines)
             {
+                // Continue if accum line count + next lines < scrolling area
+                if (accumulatedLineCount + line.WrappedText.Count < TopLineIndex) {
+                    accumulatedLineCount += line.WrappedText.Count;
+                    continue;
+                }
+                // Break if exceeds scrolling area
+                if (accumulatedLineCount - TopLineIndex > Console.BufferHeight - 3) break;
```

Comment:
Can you extract a named constant for `3` here to indicate what it means?

Reply:
Actually, looking at later uses, maybe pull out a property for `Console.BufferHeight - 3`?